src/panel/direction/DirectionPanel.jsx

```diff
@@ -80,6 +83,35 @@ export default class DirectionPanel extends React.Component {
     document.body.classList.remove('directions-open');
   }
 
+  getAddresses = (origin, destination) => {
+    const inputTextPromises = [
+      origin
+        ? origin.type === 'latlon'
+          ? IdunnPoi.poiApiLoad(origin)
+          : Promise.resolve(origin)
+        : null,
+      destination
+        ? destination.type === 'latlon'
+          ? IdunnPoi.poiApiLoad(destination)
+          : Promise.resolve(destination)
+        : null,
+    ];
+    Promise.all(inputTextPromises).then(([ origin, destination ]) => {
```

Comment:
If I'm not wrong, there is no need for this forced synchronization. Any of two points text can be updated independantly from the other, as soon as we get the response.